mtags/src/main/scala-3.1/scala/meta/internal/pc/MetalsDealising.scala

```diff
@@ -0,0 +1,19 @@
+package scala.meta.internal.pc
+
+import dotty.tools.dotc.core.Contexts.Context
+import dotty.tools.dotc.core.Types.AppliedType
+import dotty.tools.dotc.core.Types.Type
+
+object MetalsDealising:
+  extension (tpe: Type)
+    def metalsDealias(keepOpaques: Boolean = false)(using Context): Type =
+      if keepOpaques
+      then tpe // in this case we don't dealias for this version
```

Comment:
```suggestion
      if keepOpaques && tpe.typeSymbol.isOpaqueAlias
      then tpe
```
wouldn't have the same result as in later versions? We wouldn need to have a different behaviour for 3.1 and 3.0

Reply:
This wouldn't work for such a case:
```
object O:
  opaque type M = Int
  type W = M => Int
  def get: W = ???

val m = O.get
```
However, to my great confusion, I don't see any difference between the results of `tpe.dealias` and `tpe.dealiasKeepOpaques`. Neither dealiases opaque types...